docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
+
+### Storage Component
+
+The following diagram shows the class diagram for `Storage`.
+
+![StorageClassDiagram](img/StorageClassDiagram.png)
+
+1. `Storage` has `save()` and `load()` methods. These are called by `InvMgr` when needed.
+2. `save(itemList)` writes the contents of an `itemList` to a file.
+3. `load(itemList)` loads a JSON file into an `itemList`.
+4. Gson is used to serialise and deserialise the `itemList` of type `ArrayList<Item>`.
+
+## Implementation
+
+### Description Command
+![DescCommandSequenceDiagram](img/DescCommandSequenceDiagram.png)
+The following diagram shows the sequence diagram for retrieving the description of an item.
+
+For a user who is unaware of what an item is about, he/she can enter the command eg. `desc 2` command to extract the description for the second item in the inventory list. This command is interpreted by the `Parser` and a `DescCommand` is returned to `InvMgr`. `InvMgr` calls the execute command of `DescCommand` which retrieves the item's information from the `ItemList` and then outputs them into the `Ui` for the user to see.
+
+### Add Command
+![AddCommandSequenceDiagram](img/AddCommandSequenceDiagram.png)
+
+The following diagram shows the sequence diagram of the addition of an item.
+
+The user starts by typing an add command. The example used in the diagram above is the addition of an item with the name `Paper Cup`, quantity of `25` and description of `100ml paper cups`. The full command is `add n/Paper Cup q/25 d/100ml paper cups`.
+
+1. The `run()` method within `InvMgr` calls the static method `parse()` in the `Parser` class, providing the entire string of input entered by the user.
+2. Within `parse()`, the string is identified to start with the word `add`, and executes the code within the case. The case attempts to create an item using the string by self-invoking `createItem()`.
+3. `createItem()` extracts the relevant arguments from the input string and generates a new `Item` which is returned to the `parse()` case block.
+4. `parse()` uses the `Item` to generate a new `AddCommand` which is returned to the `run()` method.
+5. The `run()` method calls on the `execute()` function in the `AddCommand` which will add the generated item to the `ItemList` using its `addItem()` method.
+6. `AddCommand` will converse with `Ui` to show a message that the item has been added. In this case, the item to add will be printed as the name of the item, followed by " has been added!".
+
+### Delete Command
+![DeleteCommandSequenceDiagram](img/DeleteCommandSequenceDiagram.png)
```

Comment:
Some of the diagrams are above the "The following diagram" line while others are below. You may consider keeping it consistent :D